client/v3/naming/doc.go

```diff
@@ -12,45 +12,48 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Package naming provides an etcd-backed gRPC resolver for discovering gRPC services.
+// Package naming provides:
+//  - subpackage endpoints: an abstraction layer to store and read endpoints
+//    information from etcd.
+//  - subpackage resolver: an etcd-backed gRPC resolver for discovering gRPC
+//    services based on the endpoints configuration
 //
 // To use, first import the packages:
 //
 //	import (
 //		"go.etcd.io/etcd/client/v3"
-//		etcdnaming "go.etcd.io/etcd/client/v3/naming"
-//
+//  	"go.etcd.io/etcd/client/v3/naming/endpoints"
+//  	"go.etcd.io/etcd/client/v3/naming/resolver"
 //		"google.golang.org/grpc"
-//		"google.golang.org/grpc/naming"
 //	)
 //
 // First, register new endpoint addresses for a service:
 //
 //	func etcdAdd(c *clientv3.Client, service, addr string) error {
-//		r := &etcdnaming.GRPCResolver{Client: c}
-//		return r.Update(c.Ctx(), service, naming.Update{Op: naming.Add, Addr: addr})
+//    em := endpoints.NewManager(c, service)
+//		return endpoints.ManagerAddEndpoint(c.Ctx(), em, service+"/"+addr, endpoints.Endpoint{Addr:addr});
 //	}
 //
 // Dial an RPC service using the etcd gRPC resolver and a gRPC Balancer:
 //
 //	func etcdDial(c *clientv3.Client, service string) (*grpc.ClientConn, error) {
-//		r := &etcdnaming.GRPCResolver{Client: c}
-//		b := grpc.RoundRobin(r)
-//		return grpc.Dial(service, grpc.WithBalancer(b))
+//     etcdResolver, err := resolver.NewBuilder(c);
+//     if err { return nil, err }
+//     return  grpc.Dial("etc://foo", grpc.WithResolvers(etcdResolver))
 //	}
 //
 // Optionally, force delete an endpoint:
 //
 //	func etcdDelete(c *clientv3, service, addr string) error {
-//		r := &etcdnaming.GRPCResolver{Client: c}
-//		return r.Update(c.Ctx(), service, naming.Update{Op: naming.Delete, Addr: "1.2.3.4"})
+//	  em := endpoints.NewManager(c, service)
+//    return endpoints.ManagerDeleteEndpoint(c.Ctx(), em, service+"/"+addr)
 //	}
 //
 // Or register an expiring endpoint with a lease:
 //
-//	func etcdLeaseAdd(c *clientv3.Client, lid clientv3.LeaseID, service, addr string) error {
-//		r := &etcdnaming.GRPCResolver{Client: c}
-//		return r.Update(c.Ctx(), service, naming.Update{Op: naming.Add, Addr: addr}, clientv3.WithLease(lid))
+//	func etcdAdd(c *clientv3.Client, lid clientv3.LeaseID, service, addr string) error {
+//    em := endpoints.NewManager(c, service)
```

Comment:
Can we fix the indentation to be consistent?

Reply:
I wanted to keep the interface minimalistic.
The benefit of `Update` is that is support multiple atomic changes in a single transaction.
`AddEndpoint` is a synthactic sugar on top of Update, so could be implemented in such a way that any implementation of the endpoint.Manager gets it for free. 
It's not that important if we imagine at most 2 implementations: real & fake. 

Let me know - happy to change.

Reply:
Done. (It was tab vs. space issue).